unity/Assets/Scripts/IK_Robot_Arm_Controller.cs

```diff
@@ -517,17 +517,35 @@ public void DropObject()
             jointMeta.rootRelativePosition = FirstJoint.InverseTransformPoint(joint.position);
 
             //local rotation currently relative to immediate parent joint
-            joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
-            jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            if (joint.rotation != joint.GetChild(0).rotation)
+            {
+                joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
+                jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
```

Comment:
If it is a Quaternion assignment use Quaternion instead of Vector4

Reply:
It's not, it's the axis of rotation, followed by the angle (magnitude).

Reply:
it is a Quaternion and will be treated as such https://docs.unity3d.com/ScriptReference/Transform-localRotation.html

Reply:
"jointMeta.localRotation" is not the same as "<transform>.localRotation". "localRotation" is simply the name we use for the custom-class "JointMetadata()" for storing the local rotation, in angle-axis notation. I'll admit the naming convention is misleading.